Add WeakHashMap to Introduction.md

rniwa opened the pull request:
#### 3f6e8909eda3320f7c4244952791c207a33f1fbd
<pre>
Add WeakHashMap to Introduction.md
<a href="https://bugs.webkit.org/show_bug.cgi?id=242583">https://bugs.webkit.org/show_bug.cgi?id=242583</a>

Reviewed by Wenson Hsieh.

Added a section about WeakHashMap.

* Introduction.md:

Canonical link: <a href="https://commits.webkit.org/252333@main">https://commits.webkit.org/252333@main</a>
</pre>

whsieh commented on Introduction.md:
```diff
@@ -715,6 +715,14 @@ Because `WeakHashSet` does not get notified when the referenced object is delete
 
+### WeakHashMap
+
+Like `WeakHashSet`, `WeakHashMap` is  specialized class to map a WeakPtr key with a value.
```
Nit - it looks like there's an extra space after "is". Maybe, "…is a specialized class…"?

rniwa replied:
oh, weird. Don't know what happened there but should be "is a specialized class" as you say.

rniwa committed:
Add WeakHashMap to Introduction.md
https://bugs.webkit.org/show_bug.cgi?id=242583

Reviewed by Wenson Hsieh.

Added a section about WeakHashMap.

* Introduction.md:

Canonical link: https://commits.webkit.org/252333@main
